doc/conf.py

```diff
@@ -325,6 +327,10 @@ def setup(app):
         'numpy': 'http://docs.scipy.org/doc/numpy-1.9.1'},
     'examples_dirs': examples_dirs,
     'gallery_dirs': gallery_dirs,
+    'subsection_order': ExplicitOrderStrict(['../examples/sin_func',
```

Comment:
I would be fine with allowing a list for simplifying the user's life. We should still allow a sort key to cover the general case. There were long discussions on this in related issues and PRs.